util/network-devp2p/src/node_table.rs

```diff
@@ -104,6 +104,10 @@ impl NodeEndpoint {
 		self.to_rlp(rlp);
 	}
 
+	pub fn is_valid_sync_node(&self) -> bool {
+		self.address.port() != 0
+	}
```

Comment:
That's not really the definition of `valid`, it should include the checks of UDP port and IP address.
Maybe have `is_valid_sync_node` and `is_valid_discovery_node` (removing `is_valid`) would help make this distinction, with `is_valid_sync_node` being `is_valid_discovery_node() && tcp_port != 0`